controller/deployments_urlprovider.go

```diff
@@ -185,6 +185,36 @@ func NewTenantURLProviderFromTenant(t *app.UserService, token string, proxyURL s
 	return newTenantURLProviderFromTenant(t, token, proxyURL)
 }
 
+// GetEnvironmentMapping returns a map whose keys are environment names, and values are the Kubernetes namespaces
+// that represent those environments
+func (up *tenantURLProvider) GetEnvironmentMapping() (map[string]string, error) {
+	result := make(map[string]string)
+	// Exclude internal namespaces where the user cannot deploy applications
+
+	// Deployments API will receive requests by environment name (e.g. "run", "stage").
+	// These names correspond to the "type" attribute in Namespaces.
+	for envNS, attr := range up.namespaces {
+		envName := attr.Type
+		if envName == nil || len(*envName) == 0 {
+			return nil, errs.Errorf("namespace %s has no type", envNS)
```

Comment:
should you return an error, or just log the error and keep iterating on the namespaces ?